### PR TITLE
docs: update operator documentation to reflect dex migration 

### LINKED ISSUE #483: Unify SSO configuration in Argo CD CR
**Is your feature request related to a problem? Please describe.**
Currently, argocd-operator supports both dex and keycloak SSO providers, but the process to configure both of them differs and there's no intuitive, standard way to configure and switch between them. 

**Describe the solution you'd like**
we should move `dex` configuration from `spec.dex` to `spec.sso` to align it with keycloak. This would allow us to have a more consistent user experience installing the different sso providers supported today and also make it extensible to other potential providers in the future

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jaideepr97**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind cleanup
> /kind failing-test
> /kind enhancement
/kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:
This PR updates the api documentation to reflect the new SSO structure (implemented in https://github.com/argoproj-labs/argocd-operator/pull/488), as well as the dex related documentation to reflect the updated, unified SSO module

NOTE: This PR should only be merged after https://github.com/argoproj-labs/argocd-operator/pull/488 has been merged

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #483 

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by jannfis** [comment]
Are you sure this is a docs change? There's loads of code changes in the commits

**Comment by jaideepr97** [comment]
Hey @jannfis 
Those code changes are here because for this PR I branched off of the branch that I used to raise https://github.com/argoproj-labs/argocd-operator/pull/488 since I needed to reference the api changes I made in https://github.com/argoproj-labs/argocd-operator/pull/488 in order to update the docs accurately 

This PR is meant to be merged after https://github.com/argoproj-labs/argocd-operator/pull/488 goes in so that only the docs changes will remain when its time to merge this PR

File: docs/usage/dex.md [hunk 1]
```diff
@@ -65,30 +67,41 @@ metadata:
...
+              clientSecret: $dex.github.clientSecret # Alternatively $<some_K8S_secret>:dex.github.clientSecret
+              orgs:
+              - name: dummy-org
 ```
 
-## Disable DEX
+## Install/Uninstall DEX
 
-Dex is installed by default for all the Argo CD instances created by the operator. You can disable this behavior using the environmental variable `DISABLE_DEX` on the operator.
+Dex is not installed by default for any of the Argo CD instances created by the operator. It can be enabled by setting `.spec.sso.provider` to `dex` and supplying a non-empty `.spec.sso.dex` section within the Argo CD custom resource. For example:
```
Comment (iam-veeramalla) [inline docs/usage/dex.md:88]: We were deploying DEX OOTB for the Argo CD instances that are created from the Operator Console.
https://github.com/argoproj-labs/argocd-operator/blob/6de5fc2585acdcb595c04cf400139fac48eda431/bundle/manifests/argocd-operator.clusterserviceversion.yaml#L50


I dont think it is a good idea to change the behavior. What do you think ?

**Review by iam-veeramalla — APPROVED** [review]
LGTM, Thanks @jaideepr97

**Pull request merged**